docs/config_entries_config_flow_handler.md

```diff
@@ -83,6 +83,37 @@ if device_unique_id:
 await self._async_handle_discovery_without_unique_id()
 ```
 
+### Unique ID Requirements
+
+A Unique ID is used to match a config entry to the underlying device or api. The Unique ID must be stable and should not be able to be changed by the user. The Unique ID can be used to update the config entry data when device access details change. For example, for devices that communicate over the local network, if the IP address changes due a new DHCP assignment, the integration can use the Unique ID to update the host using the following code snippet:
```

Comment:
- `api` -> `API`
- `due a new DHCP` -> `due to a new DHCP`

Reply:
https://github.com/home-assistant/developers.home-assistant/pull/834